src/crypto.cpp

```diff
@@ -265,29 +232,26 @@ int Crypto::AsymDec(
     *data_size = 0;
 
     // Decrypt each ciphertext in `enc_data`
-    for (size_t i = 0; i < enc_data_size / RSA_MOD_SIZE; i++) {
+    size_t num_cts = enc_data_size / RSA_MOD_SIZE;
+    for (size_t i = 0; i < num_cts; i++) {
         // The size of the current plaintext
         size_t pt_size = 0;
 
-        // The sequence number for the current ciphertext
-        std::vector<uint8_t> seq(sizeof(i));
-        memcpy(seq.data(), &i, sizeof(i));
+        // The AAD for each ciphertext is "{num_cts}||{seq}"
+        std::ostringstream aad_stream;
+        aad_stream << num_cts << "||" << i;
+        std::string aad_str = aad_stream.str();
+        std::vector<uint8_t> aad(aad_str.begin(), aad_str.end());
 
         // Decrypt the ciphertext.
         res = mbedtls_rsa_rsaes_oaep_decrypt(
-            rsa_context,
-            mbedtls_ctr_drbg_random,
-            &m_ctr_drbg_context,
-            MBEDTLS_RSA_PRIVATE,
-            seq.data(),
-            seq.size(),
-            &pt_size,
-            next_ct,
-            next_pt,
-            RSA_MOD_SIZE);
+            rsa_context, mbedtls_ctr_drbg_random, &m_ctr_drbg_context,
```

Comment:
has this been formatted? i think Chromium style clang format specifies that each argument is on a different line

Reply:
Ah, there was actually a problem with the `.clang_format` we were using. This will need to be updated for the client as well (I added a story)